web-server/src/components/WorkflowSelector.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+  Checkbox,
+  Chip,
+  FormControl,
+  InputLabel,
+  ListItemText,
+  MenuItem,
+  OutlinedInput,
+  Select,
+  CircularProgress
+} from '@mui/material';
+import { union } from 'ramda';
+import { FC, useCallback, useMemo } from 'react';
+
+import { handleApi } from '@/api-helpers/axios-api-instance';
+import { Line } from '@/components/Text';
+import { track } from '@/constants/events';
+import { CIProvider, Integration } from '@/constants/integrations';
+import { useAuth } from '@/hooks/useAuth';
+import { useEasyState, useBoolState } from '@/hooks/useEasyState';
+import {
+  BaseRepo,
+  RepoWorkflowResponse,
+  RepoWorkflow
+} from '@/types/resources';
+import { depFn } from '@/utils/fn';
+
+import { AsyncSelectOptions } from './AsyncSelect';
+import { FlexBox } from './FlexBox';
+import { useTeamCRUD } from './Teams/useTeamsConfig';
+
+export const DeploymentWorkflowSelector: FC<{ repo: BaseRepo }> = ({
+  repo
+}) => {
+  const { orgId } = useAuth();
+
+  const options = useEasyState<AsyncSelectOptions>([]);
+  const nextPageToken = useEasyState<string | null>('');
+  const loading = useBoolState();
+  const { updateWorkflowsForTeam } = useTeamCRUD();
+  const provider = Integration.GITHUB;
+
+  const selectedOptions = useMemo(
+    () =>
+      repo.repo_workflows?.map((val) => ({
+        label: val?.name,
+        value: val?.value
```

Comment:
Not a change:

When can the value of repo_workflows list items be falsy to need optional chaining here?

Reply:
removed